src/System.Management.Automation/DscSupport/JsonCimDSCParser.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Security;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal.CrossPlatform
+{
+    /// <summary>
+    /// Class that does high level Cim schema parsing.
+    /// </summary>
+    internal class CimDSCParser
+    {
+        private readonly JsonDeserializer _jsonDeserializer;
+        
+        internal CimDSCParser()
+        {
+            _jsonDeserializer = JsonDeserializer.Create();
+        }
+
+        internal IEnumerable<PSObject> ParseSchemaJson(string filePath, bool useNewRunspace = false)
+        {
+            try
+            {
+                string json = File.ReadAllText(filePath);
+                string fileNameDefiningClass = Path.GetFileNameWithoutExtension(filePath);
+                int dotIndex = fileNameDefiningClass.IndexOf(".schema", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Actual file names that are parsed with this may or may not have `.schema` in their name: like `MyBaseClass.schema.json` or `MyBaseClass.json` . In both cases, per regulation, the part before the first dot is supposed to match the class definition inside the file; so this code just tries to extract the first part (before the first dot).